front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy

```diff
@@ -44,6 +44,18 @@ class PipelineController {
     pipelineDAO.getPipelinesByApplication(application)
   }
 
+  @RequestMapping(value = 'getById/{id:.+}', method = RequestMethod.GET)
```

Comment:
Yup, I think that's fine on both counts (having a v2 and non-v2
api/controller).

Eventually the v2 api will be updated to include the missing functionality
from v1, and v1 becomes deprecated. Front50 is an internal API and I
wouldn't expect direct calls to it from anything but other Spinnaker
services (it should either require two-way ssl or be security
group/firewalled).

You're absolutely correct on the last part, would need to supply both
application and name query parameters. I wouldn't have a problem if the API
supported application, name, application + name or only application,
application + name (and raised a non-200 if only name were provided). In
the former, if you just supplied name, you would get the similarly named
pipelines across applications. Not particularly common.

Doing this is a nice step forward towards a more sensible API and the
patterns can be applied to the other types (applications, notifications,
etc.).


On Thu, Jun 23, 2016 at 6:11 AM Eric Wiseblatt <notifications@github.com>
wrote:

> In
> front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineController.groovy
> <https://github.com/spinnaker/front50/pull/98#discussion_r68230998>:
>
> > @@ -44,6 +44,18 @@ class PipelineController {
> >      pipelineDAO.getPipelinesByApplication(application)
> >    }
> >
> > +  @RequestMapping(value = 'getById/{id:.+}', method = RequestMethod.GET)
>
> Ok. Reset. A couple questions.
> /v2/pipelines would add a new controller. Is that ok? Swagger wants them
> to have different names (even though they are in different packages). I
> called it a V2PipelineController. Now the V2 controller has a couple
> methods and the old controller has the rest. Its fine functionally, but the
> API is odd, at least when going through the swagger web ui. I dont have
> stake in that so dont care, but want to verify.
>
> Also, "name" is the pipeline name. Names are scoped to applications. So if
> you are providing a name, dont you also need the application?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/front50/pull/98/files/15c62c9b564dc818eecfd59c4d45ef4a0889b2bf#r68230998>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAXuLEQcO9Z443Wc4Uzq_Re337BFySWLks5qOoYTgaJpZM4I58yx>
> .
>